### PR TITLE
Fixed fatal error for Event Participants custom search if price set field option is disabled after a registrant has already selected it.

### PR DESCRIPTION
**Pull request by pradpnayak**
Overview
----------------------------------------
From https://civicrm.stackexchange.com/questions/30602/unable-to-use-price-set-details-for-event-participants-custom-search-if-price-se

Before
----------------------------------------
Fatal error

After
----------------------------------------
No Fatal error

### CODE REVIEW

File: CRM/Contact/Form/Search/Custom/PriceSet.php [hunk 1]
```diff
@@ -116,19 +116,24 @@ public function fillTable() {
...
+          ON p.contact_id = c.id AND c.is_deleted = 0
+        INNER JOIN civicrm_line_item l
+          ON p.id = l.entity_id AND l.entity_table ='civicrm_participant'
+        INNER JOIN civicrm_price_field_value cpfv
+          ON cpfv.id = l.price_field_value_id AND cpfv.is_active = 1
+        INNER JOIN civicrm_price_field cpf
+          ON cpf.id = l.price_field_id AND cpf.is_active = 1
+        INNER JOIN civicrm_price_set cps
+          ON cps.id = cpf.price_set_id AND cps.is_active = 1
+      WHERE  p.event_id = {$this->_eventID}
```
Comment (seamuslee001) [inline CRM/Contact/Form/Search/Custom/PriceSet.php]: @	pradpnayak could `$this->_eventID` be set by URL at all?

Comment (eileenmcnaughton) [reply CRM/Contact/Form/Search/Custom/PriceSet.php]: It should be parameterised anyway I guess - it's on the form layer & the answer might be no now...

Comment (freeform-sg) [reply CRM/Contact/Form/Search/Custom/PriceSet.php]: sorry for delay in testing, finally got a chance to take a look, I received a syntax error message:

```
[nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'INNER JOIN civicrm_participant p
          ON p.contact_id = c.id AND c.is_delet' at line 6]
```

My SQL isn't very strong, but I believe you just need to drop the comma before the first inner join.

```
      FROM civicrm_contact c**,**
        INNER JOIN civicrm_participant p
```

**Comment by freeform-sg** [comment]
Thanks for this! Sorry for delay in testing, finally got a chance to take a look. I now get "DB Error: syntax error", from the error log:

```
[nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'INNER JOIN civicrm_participant p
          ON p.contact_id = c.id AND c.is_delet' at line 6]
```

My SQL isn't very strong, but I believe you just need to drop the comma before the first inner join (tested this theory and error went away).

```
      FROM civicrm_contact c**,**
        INNER JOIN civicrm_participant p
```

**Comment by eileenmcnaughton** [comment]
@pradpnayak I think we need to get a test on any changes to search functionality

**Comment by freeform-sg** [comment]
Hi there - was this ever merged with master? If it was then there may be a regression. We are still requiring this patch with the latest version of PriceSet.php.

**Comment by eileenmcnaughton** [comment]
I'm not sure it was - we try not to make any changes to custom searches without adding a unit test & it looked like one didn't get written for this

**Comment by freeform-sg** [comment]
Ok, thank you, that makes sense. Does it being marked closed mean that's not going to happen? Is it possible to put it back into needs-test? The patch still works with version 5.23.0.

**Comment by eileenmcnaughton** [comment]
So normally if there is a bug but the  PR is not review-ready  we  would track it through creating an issue in gitlab. I think on balance on this one if you  have been running it successfully in production all this time &  it's a very low-usage/ unloved corner of Civi I'm going to re-open this & merge it based on your testing (& let it slip the test expectation). The scope of the patch is very narrow so it shouldn't affect anything else

**Comment by freeform-sg** [comment]
Thanks a bunch, you rock!

**Comment by seamuslee001** [comment]
Test failure `CRM_Contact_Form_Search_Custom_PriceSetTest.testRunSearch` relates, I have put up a reivewers cut of this in https://github.com/civicrm/civicrm-core/pull/16894 and took the opportunity ot parametise the queries

**Pull request closed without merging**